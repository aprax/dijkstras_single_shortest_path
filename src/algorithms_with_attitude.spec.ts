import findShortestPath from ".";
import type { Nodes } from './types';

it('should find the shortest paths to all other nodes from start', () => {
	const nodes: Nodes = new Map([
		['Start', [['A', 8], ['D', 2], ['G', 3], ['F', 7]]],
		['A', [['B', 3]]],
		['B', [['A', 1], ['C', 4], ['E', 0]]],
		['C', [['B', 0]]],
		['D', [['E', 13], ['F', 3]]],
		['E', [['B', 1], ['C', 0], ['F', 5]]],
		['F', [['C', 8], ['D', 4]]],
		['G', [['F', 5]]]
	])
	expect(findShortestPath('Start', nodes)).toMatchSnapshot()
})