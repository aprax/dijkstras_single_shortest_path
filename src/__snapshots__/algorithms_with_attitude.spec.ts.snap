// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should find the shortest paths to all other nodes from start 1`] = `
Map {
  "Start" => Array [
    "Start",
    0,
  ],
  "A" => Array [
    "Start",
    8,
  ],
  "D" => Array [
    "Start",
    2,
  ],
  "G" => Array [
    "Start",
    3,
  ],
  "F" => Array [
    "D",
    5,
  ],
  "B" => Array [
    "A",
    11,
  ],
  "E" => Array [
    "B",
    11,
  ],
  "C" => Array [
    "E",
    11,
  ],
}
`;
